web/src/components/storage/iscsi/TargetsSection.jsx

```diff
@@ -0,0 +1,163 @@
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useReducer } from "react";
+import {
+  Button,
+  Toolbar, ToolbarItem, ToolbarContent,
+  Skeleton
+} from "@patternfly/react-core";
+
+import { Section } from "~/components/core";
+import { NodesPresenter, DiscoverForm } from "~/components/storage/iscsi";
+import { useInstallerClient } from "~/context/installer";
+import { useCancellablePromise } from "~/utils";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "SET_NODES": {
+      return { ...state, nodes: action.payload.nodes };
+    }
+
+    case "ADD_NODE": {
+      if (state.nodes.find(n => n.id === action.payload.node.id)) return state;
+
+      const nodes = [...state.nodes];
+      nodes.push(action.payload.node);
+      return { ...state, nodes };
+    }
+
+    case "UPDATE_NODE": {
+      const index = state.nodes.findIndex(n => n.id === action.payload.node.id);
+      if (index === -1) return state;
+
+      const nodes = [...state.nodes];
+      nodes[index] = action.payload.node;
+      return { ...state, nodes };
+    }
+
+    case "REMOVE_NODE": {
+      const nodes = state.nodes.filter(n => n.id !== action.payload.node.id);
+      return { ...state, nodes };
+    }
+
+    case "OPEN_DISCOVER_FORM": {
+      return { ...state, isDiscoverFormOpen: true };
+    }
+
+    case "CLOSE_DISCOVER_FORM": {
+      return { ...state, isDiscoverFormOpen: false };
+    }
+
+    case "START_LOADING": {
+      return { ...state, isLoading: true };
+    }
+
+    case "STOP_LOADING": {
+      return { ...state, isLoading: false };
+    }
+
+    default: {
+      return state;
+    }
+  }
+};
+
+const initialState = {
+  nodes: [],
+  isDiscoverFormOpen: false,
+  isLoading: true
+};
+
+export default function TargetsSection() {
+  const { storage: client } = useInstallerClient();
+  const { cancellablePromise } = useCancellablePromise();
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    const loadNodes = async () => {
+      dispatch({ type: "START_LOADING" });
+      const nodes = await cancellablePromise(client.iscsi.getNodes());
+      dispatch({ type: "SET_NODES", payload: { nodes } });
+      dispatch({ type: "STOP_LOADING" });
+    };
+
+    loadNodes().catch(console.error);
+  }, [cancellablePromise, client.iscsi]);
+
+  useEffect(() => {
+    const action = (type, node) => dispatch({ type, payload: { node } });
+
+    client.iscsi.onNodeAdded(n => action("ADD_NODE", n));
+    client.iscsi.onNodeChanged(n => action("UPDATE_NODE", n));
+    client.iscsi.onNodeRemoved(n => action("REMOVE_NODE", n));
+  }, [client.iscsi]);
+
+  const openDiscoverForm = () => {
+    dispatch({ type: "OPEN_DISCOVER_FORM" });
+  };
+
+  const closeDiscoverForm = () => {
+    dispatch({ type: "CLOSE_DISCOVER_FORM" });
+  };
+
+  const SectionContent = () => {
+    if (state.isLoading) return <Skeleton />;
```

Comment:
Here I prefer `SectionSkeleton`. We are aware that it has to be polished, but please use it by now when you're waiting for the section content. The _raw_ Skeleton inside a row when waiting for its content is ok.